app/src/main/java/com/sebastianvm/mango/model/PayFrequency.kt

```diff
@@ -0,0 +1,86 @@
+package com.sebastianvm.mango.model
+
+import com.sebastianvm.mango.util.date.DateRange
+import kotlinx.datetime.LocalDate
+import kotlinx.serialization.Serializable
+
+/**
+ *
+ * Pay period: A pay period is a time frame used to calculate earned wages and determine when
+ * employees receive their paychecks. Pay periods are fixed and most often recurring on a weekly,
+ * bi-weekly, semi-monthly or monthly basis.
+ *
+ * Pay Date: The employee does not receive his or her check at the end of the pay period, but a
+ * number of days after it has concluded, which is known as the pay date.
+ *
+ *  [Source](https://www.netsuite.com/portal/resource/articles/human-resources/pay-period.shtml)
+ *
+ *  For these, we include pay period and pay day information:
+ *  @property firstPayPeriod Defines the range of the first pay period during which a person is
+ *  paid. This can then be used to calculate all the following pay periods.
+ *
+ *
+ *  @property daysFromEndOfPayPeriodToPayDay Used to determine when a person is actually paid for
+ *  pay period.
+ *
+ *
+ *  @property lastDay The last day a person earns the income. Note this is not the last day the
+ *  person will get paid. That day will be the last day of the current pay period + [daysFromEndOfPayPeriodToPayDay]
+ *
+ */
+@Serializable
+sealed interface PayFrequency {
```

Comment:
wow does kotlinx-serialization just do this?

Reply:
What exactly, support sealed interfaces? If so yes, you just need to annotate the sealed interface and all subclasses